ecommerce_worker/utils.py

```diff
@@ -46,19 +49,32 @@ def get_configuration(variable, site_code=None):
     return setting_value
 
 
-def get_ecommerce_client(url_postfix='', site_code=None):
+def get_access_token():
     """
-    Get client for fetching data from ecommerce API.
-    Arguments:
-        site_code (str): (Optional) The SITE_OVERRIDES key to inspect for site-specific values
-        url_postfix (str): (Optional) The URL postfix value to append to the ECOMMERCE_API_ROOT value.
+    Returns an access token for this site's service user.
 
     Returns:
-        EdxRestApiClient object
+        str: JWT access token
     """
-    ecommerce_api_root = get_configuration('ECOMMERCE_API_ROOT', site_code=site_code)
-    signing_key = get_configuration('JWT_SECRET_KEY', site_code=site_code)
-    issuer = get_configuration('JWT_ISSUER', site_code=site_code)
-    service_username = get_configuration('ECOMMERCE_SERVICE_USERNAME', site_code=site_code)
-    return EdxRestApiClient(
-        ecommerce_api_root + url_postfix, signing_key=signing_key, issuer=issuer, username=service_username)
+    oauth_access_token_url = urljoin(get_configuration('BACKEND_SERVICE_EDX_OAUTH2_PROVIDER_URL'), 'access_token/')
+    try:
+        response = requests.post(
+            oauth_access_token_url,
+            data={
+                'grant_type': 'client_credentials',
+                'client_id': get_configuration('BACKEND_SERVICE_EDX_OAUTH2_KEY'),
+                'client_secret': get_configuration('BACKEND_SERVICE_EDX_OAUTH2_SECRET'),
+                'token_type': 'jwt',
+            },
+            headers={
+                'User-Agent': 'ecommerce-worker',
+            },
+            timeout=(REQUEST_CONNECT_TIMEOUT, REQUEST_READ_TIMEOUT)
+        )
+        response.raise_for_status()
+        data = response.json()
+        access_token = data['access_token']
+    except KeyError as exc:
```

Comment:
Please handle the HTTPError exception instead. As for the access_token, what should be the default return in case the response is successfully got but key 'access_token' is not in the `data`?

Reply:
decided leave exception handling